chmod rootfs files when mounting.

SeanTAllen opened the pull request:
This fixes #236.

The problem was that we were setting "/" to be 700 for perms which doesn't allow any user other than root to access "/tmp" and other directories.

After some review and conversation with @prp, I believe that in general the existing code in lkl_prepare_rootfs was there as  "just in case" sanity check. It creates directories if they are missing and provides some permissions (which differ from normal Linux installs). However, those directories almost always existed.

This change will now chmod those directories (including "/") if they exit. Additionally it updates the perms to be set to be 755 to match a standard Linux environment.

SeanTAllen commented:
This will need to be rebased when https://github.com/lsds/sgx-lkl/pull/724 is merged.

At the moment, sgx-lkl is behind on ltp tests so there are a few tests included in this PR that aren't part of the changes. Once 724 is merged, that will go away.

PR for ltp changes is here:

https://github.com/lsds/ltp/pull/50

hukoyu reviewed (approved):
LGTM. @SeanTAllen  thanks for finding the root cause of this issue and fixing it.

Commit:
chmod rootfs files when mounting

This fixes #236.

The problem was that we were setting "/" to be 700 for perms which
doesn't allow any user other than root to access "/tmp" and
other directories.

After some review and conversation with @prp, I believe that in general
the existing code in lkl_prepare_rootfs was there as a "just in case"
sanity check. It creates directories if they are missing and provides
some permissions (which differ from normal Linux installs). However,
those directories almost always existed.

This change will now chmod those directories (including "/") if they
exit. Additionally it updates the perms to be set to be 755 to match
a standard Linux environment.

Commit:
Include tests from LTP

Commit:
Remove debugging statement

wintersteiger reviewed (approved):
Looks good to me! I'm not sure whether 0755 is the right choice - I hope we won't need to add new config settings for these permissions, if applications expect to have access when they don't.
